.travis/pulp-operator-check-and-wait.sh

```diff
@@ -144,62 +144,40 @@ for tries in {0..120}; do
   fi
 done
 
-for tries in {0..120}; do
-  if [[ $tries -eq 120 ]]; then
-    echo "ERROR 5: pulp-api never connected to the database"
-    storage_debug
-    echo "$output"
-    exit 5
-  fi
-  output=$(http --timeout 5 --check-status --pretty format --print hb $URL 2>&1)
-  if [[ "$(echo "$output" | sed -ne '/{/,$ p' | jq -r .database_connection.connected)" = "true" ]]; then
-    echo "pulp-api is connected to the database"
-    break
-  fi
-  sleep 5
-done
+messages=(
+    "pulp-api is connected to the database"
+    "pulp-api is connected to redis"
+    "Content app is online"
+    "1 or more worker is online"
+)
 
-for tries in {0..120}; do
-  if [[ $tries -eq 120 ]]; then
-    echo "ERROR 6: pulp-api never connected to redis"
-    storage_debug
-    echo "$output"
-    exit 6
-  fi
-  output=$(http --timeout 5 --check-status --pretty format --print hb $URL 2>&1)
-  if [[ "$(echo "$output" | sed -ne '/{/,$ p' | jq -r .redis_connection.connected)" = "true" ]]; then
-    echo "pulp-api is connected to redis"
-    break
-  fi
-  sleep 5
-done
+error_messages=(
+    "ERROR 5: pulp-api never connected to the database"
+    "ERROR 6: pulp-api never connected to redis"
+    "ERROR 7: Content app never came online"
+    "ERROR 8: Worker(s) never came online"
+)
 
-for tries in {0..120}; do
-  if [[ $tries -eq 120 ]]; then
-    echo "ERROR 7: Content app never came online"
-    storage_debug
-    echo "$output"
-    exit 7
-  fi
-  output=$(http --timeout 5 --check-status --pretty format --print hb $URL 2>&1)
-  if [[ "$(echo "$output" | sed -ne '/{/,$ p' | jq -r .online_content_apps)" != "[]" ]]; then
-    echo "Content app is online"
-    break
-  fi
-  sleep 5
-done
+tests=(
+    "$(echo "$output" | sed -ne '/{/,$ p' | jq -r .database_connection.connected)" = "true"
+    "$(echo "$output" | sed -ne '/{/,$ p' | jq -r .redis_connection.connected)" = "true"
+    "$(echo "$output" | sed -ne '/{/,$ p' | jq -r .online_content_apps)" != "[]"
+    "$(echo "$output" | sed -ne '/{/,$ p' | jq -r .online_workers)" != "[]"
+)
 
-for tries in {0..120}; do
-  if [[ $tries -eq 120 ]]; then
-    echo "ERROR 8: Worker(s) never came online"
-    storage_debug
-    echo "$output"
-    exit 8
-  fi
-  output=$(http --timeout 5 --check-status --pretty format --print hb $URL 2>&1)
-  if [[ "$(echo "$output" | sed -ne '/{/,$ p' | jq -r .online_workers)" != "[]" ]]; then
-    echo "1 or more worker is online"
-    break
-  fi
-  sleep 5
+for iteration in {5..8};do
+    for tries in {0..120}; do
+    if [[ $tries -eq 120 ]]; then
+        echo $error_messages[$iteration]
+        storage_debug
+        echo "$output"
+        exit $iteration
+    fi
+    output=$(http --timeout 5 --check-status --pretty format --print hb $URL 2>&1)
+    if [[ "$tests[$iteration]" ]]; then
+        echo $messages[$iteration]
```

Comment:
You need curly brackets to access items in arrays. The output on the CI demonstrates this.

Try ${error_messages[$iteration]} for example.

Also, wouldn't the first element in the list be 1, not 5?

Reply:
because from 1 to 4, it had some variations that I didn't know how to adapt, so I started with 5

Reply:
To clarify, I understand why you would want to start at 5. But it isn't working. The output like "pulp-api is connected to the database" is not showing up. And it is not showing up because you have arrays like `messages`. The 1st element in the list is 1, not 5. In fact, I suspect the tests are not actually being run either, since they are in an array as well.

Reply:
my bad! Thanks Mike, I haven't noticed the problem with array indexes. I believe I've fixed it now